instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/ClientAddressAndPortExtractor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.http;
+
+import static io.opentelemetry.instrumentation.api.instrumenter.http.ForwardedHeaderParser.extractClientIpFromForwardedForHeader;
+import static io.opentelemetry.instrumentation.api.instrumenter.http.ForwardedHeaderParser.extractClientIpFromForwardedHeader;
+
+import io.opentelemetry.instrumentation.api.instrumenter.network.internal.FallbackAddressPortExtractor;
+
+final class ClientAddressAndPortExtractor<REQUEST>
+    implements FallbackAddressPortExtractor<REQUEST> {
+
+  private final HttpServerAttributesGetter<REQUEST, ?> getter;
+
+  ClientAddressAndPortExtractor(HttpServerAttributesGetter<REQUEST, ?> getter) {
+    this.getter = getter;
+  }
+
+  @Override
+  public void extract(AddressPortSink sink, REQUEST request) {
+    // try Forwarded
+    String forwarded =
+        HttpCommonAttributesExtractor.firstHeaderValue(
+            getter.getHttpRequestHeader(request, "forwarded"));
+    if (forwarded != null) {
+      forwarded = extractClientIpFromForwardedHeader(forwarded);
```

Comment:
This is nitpicky/specific/uncommon -- BUT, hear me out, if there just so happen to be two `forwarded` headers and the first one doesn't contain the client IP but the second one does, then this won't work, right? Does this matter?

Reply:
Oh, that's a nice edge case. I'd like to make it a part of #9009 though, since this particular class will be changed significantly as part of that ticket.